src/cpp/src/sequence_group.hpp

```diff
@@ -133,7 +134,16 @@ class Sequence {
         return score;
     }
 
-    // Each KV block can be uniquely identified by 
+
+    void register_token_eviction(size_t num_evicted_tokens) {
+        m_num_evicted_tokens += num_evicted_tokens;
```

Comment:
should we just decrease context length? and perform re-rotation of KV blocks

I supposed after eviction we will have short context length as well, won't we?

Reply:
See as above. Note that currently I don't rerotate blocks (i.e. am not rewriting block contents), I fear that SW implementation will be too slow right now. Should estimate the accuracy impact with the current non-rotated implementation first, maybe it will be OK as-is for now.